device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh

```diff
@@ -1,73 +1,79 @@
+            sudo ip netns exec asic$ASIC ip link set $TEMP name $NEW # rename too final interface name
+            sudo ip netns exec asic$ASIC ip link set dev $NEW mtu 9100
+            sudo ip netns exec asic$ASIC ip link set $NEW up
         done
     done
 
     # Connect all backend namespaces to frontend namespaces
     for BACKEND in `seq $FIRST_BACKEND_ASIC $LAST_BACKEND_ASIC`; do
         for FRONTEND in `seq $FIRST_FRONTEND_ASIC $LAST_FRONTEND_ASIC`; do
-            for LINK in `seq 1 8`; do
-		FRONT_NAME="eth$((8 * $(($BACKEND - $FIRST_BACKEND_ASIC)) + $LINK + 16))"
-		BACK_NAME="eth$((8 * $FRONTEND + $LINK))"
-		echo "$FRONTEND:$FRONT_NAME - $BACKEND:$BACK_NAME"
+            for LINK in `seq 1 2`; do
+                FRONT_NAME="eth$((2 * $(($BACKEND - $FIRST_BACKEND_ASIC)) + $LINK + 4))"
+                BACK_NAME="eth$((2 * $FRONTEND + $LINK))"
+                echo "$FRONTEND:$FRONT_NAME - $BACKEND:$BACK_NAME"
                 TEMP_BACK="ethBack999"
                 TEMP_FRONT="ethFront999"
-                
                 ip link add $TEMP_BACK type veth peer name $TEMP_FRONT # temporary name to prevent conflicts between interfaces
                 ip link set dev $TEMP_BACK netns asic$BACKEND
-                ip link set dev $TEMP_FRONT netns asic$FRONTEND 
-    
+                ip link set dev $TEMP_FRONT netns asic$FRONTEND
+
                 sudo ip netns exec asic$BACKEND ip link set $TEMP_BACK name $BACK_NAME
                 sudo ip netns exec asic$FRONTEND ip link set $TEMP_FRONT name $FRONT_NAME
 
+                sudo ip netns exec asic$BACKEND ip link set dev $BACK_NAME mtu 9100
                 sudo ip netns exec asic$BACKEND ip link set $BACK_NAME up
+                sudo ip netns exec asic$FRONTEND ip link set dev $FRONT_NAME mtu 9100
                 sudo ip netns exec asic$FRONTEND ip link set $FRONT_NAME up
             done
         done
     done
 }
-
 stop() {
     for ASIC in `seq $FIRST_FRONTEND_ASIC $LAST_FRONTEND_ASIC`; do
-        for NUM in `seq 1 16`; do
+        for NUM in `seq 1 4`; do
             TEMP="eth999"
-            OLD="eth$((16 * $ASIC + $NUM))"
-            NAME="eth$((16 * $ASIC + $NUM - 1))"
+            OLD="eth$(($NUM))"
+            OLD="eth$(($NUM))"
+            NAME="eth$((4 * $ASIC + $NUM))"
             sudo ip netns exec asic$ASIC ip link set dev $OLD down
             sudo ip netns exec asic$ASIC ip link set dev $OLD name $TEMP
             sudo ip netns exec asic$ASIC ip link set dev $TEMP netns 1
             ip link set dev $TEMP name $NAME
             ip link set dev $NAME up
+            echo "$ASIC: $OLD --> $NAME"
         done
     done
 
     for ASIC in `seq $FIRST_BACKEND_ASIC $LAST_BACKEND_ASIC`; do
-        for NUM in `seq 1 $NUM_INTERFACES_PER_ASIC`; do
+        for NUM in `seq 1 4`; do
             sudo ip netns exec asic$ASIC ip link set dev eth$NUM down
             sudo ip netns exec asic$ASIC ip link delete dev eth$NUM
+            echo "$ASIC: eth$NUM"
```

Comment:
do we need this echo statement?

Reply:
We don't need this echo statement, removed echo statements added for debugging.